sdk/src/logs/logger.cc

```diff
@@ -36,23 +51,76 @@ void Logger::log(const opentelemetry::logs::LogRecord &record) noexcept
 
   // TODO: Sampler logic (should include check for minSeverity)
 
-  /**
-   * Convert the LogRecord to the heap first before sending to processor.
-   * TODO: Change the API log(LogRecord) function to log(*LogRecord) so the following line
-   * converting record a heap variable can be removed
-   */
-  auto record_pointer =
-      std::unique_ptr<opentelemetry::logs::LogRecord>(new opentelemetry::logs::LogRecord(record));
+  auto recordable = processor->MakeRecordable();
+  if (recordable == nullptr)
+  {
+    return;
+  }
+
+  // Populate recordable fields
+  if (timestamp == this->default_timestamp_)
```

Comment:
Related to my other comment, that inline timestamp when we logged - would be different from when we populate it here, since this assignment actually has to wait for the lock on `logger_provider_.lock()` and a memory allocation. In case of a congested system it might be a few milliseconds difference.

Reply:
That's a good point, I hadn't thought of that. Would getting the actual timestamp as early as the API be a time-consuming task though? If the LogRecord were filtered out by the Sampler in the SDK for example, would the previous timestamp call have been wasted/unecessary?

Reply:
It depends. In some scenarios you'd rather prefer the log time as close to original as possible, as you need the actual time when something happened - not when you got thru the lock to assign some timestamp in the SDK, esp. given it's under some lock. This is not a blocker for your PR. I'm thinking maybe we should have a method added in API later on that allows to customize this behavior: whether it's the API stamps the `now` time _right away_ , or if API stamps the indicator `please stamp the time for me` (like you are doing it here), and then later on the next stage stamps it.